autoarena/api/utils.py

```diff
@@ -0,0 +1,18 @@
+from typing import TypeVar, AsyncIterator
+
+from pydantic import RootModel
+from starlette.responses import StreamingResponse
+
+T = TypeVar("T")  # should be a Pydantic dataclass
+
+
+async def as_sse_stream(object_stream: AsyncIterator[T]) -> AsyncIterator[str]:
+    async for obj in object_stream:
+        obj_json = RootModel[T](obj).model_dump_json()
+        yield f"data: {obj_json}\n\n"
+
+
+# TODO: figure out a way to gracefully terminate this when the server is exiting
```

Comment:
This is a classic `asyncio` case where graceful shutdown is way, way harder to get than it should be. Currently, when there are active connections (e.g. `/tasks/has-active`, which will be open whenever you have the app open in your browser) the app will wait a second after Ctrl+C then kill the tasks. I exceeded my timebox hunting this down so leaving cleaning this up as a future TODO.